packages/vscode-extension/src/extension/decorators.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import * as vscode from "vscode";
+import { LanguageClient } from "vscode-languageclient/node.js";
+import { Settings } from "./settings";
+import { Range } from "vscode-languageserver-types";
+
+let skippedCodeDecorationType: vscode.TextEditorDecorationType;
+const skippedRangesByUri = new Map<string, vscode.Range[]>();
+
+export function registerSkipDecoratorType(
+  client: LanguageClient,
+  settings: Settings,
+) {
+  updateSkipDecoratorType(settings);
+
+  client.onNotification(
+    "pli/skippedCode",
+    (params: { uri: string; ranges: Range[] }) => {
+      if (!settings.skippedCodeEnabled) {
+        return;
+      }
+
+      const editor = vscode.window.visibleTextEditors.find(
+        (e) => e.document.uri.toString() === params.uri,
+      );
+      if (editor) {
```

Comment:
Suggestion: Since multiple editors are possible for the same file, you should use `filter` instead. The decoration can then be applied to every editor instead of just the first.